### PR TITLE
Create a `Report` rust wrapper around `sgx_report_t`

### PR DESCRIPTION
**Pull request by nick-mobilecoin**

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #91** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/91" target="_blank">Graphite</a> 
    * **PR #86** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/86" target="_blank">Graphite</a> 
      * **PR #92** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/92" target="_blank">Graphite</a> 
        * **PR #103** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/103" target="_blank">Graphite</a> 
          * **PR #110** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/110" target="_blank">Graphite</a> 
            * **PR #111** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/111" target="_blank">Graphite</a> 
              * **PR #112** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/112" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/112?utm_source=stack-comment).

File: core/types/src/report.rs [hunk 1]
```diff
@@ -2,15 +2,35 @@
...
+#[derive(Debug, Clone, Hash, PartialEq, Eq)]
+#[repr(transparent)]
+pub struct Mac(sgx_mac_t);
+
+new_type_accessors_impls! {
+    Mac, sgx_mac_t;
+}
+
+impl Mac {
+    pub const SIZE: usize = SGX_MAC_SIZE;
```
Comment (jcape) [inline core/types/src/report.rs:25]: Same Q as the other PR: should this be public, or can we keep it private?

**Review by jcape — APPROVED** [review]

File: core/types/src/report.rs (hunk 1 above)
Comment (nick-mobilecoin) [reply core/types/src/report.rs]: Removed it. Along with being able to derive Default it was over exposing API members thinking users might need to know how to construct a key. At this time not seeing a need for a user to write or generate a MAC, only to verify one.

**Review by awygle — APPROVED** [review]
This all seems fine.

**Comment by nick-mobilecoin** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/112?utm_source=stack-comment).


Job Reference: C6ZyZ0Ggec8V0ksW3mYN

**Pull request merged**